### PR TITLE
[UR][OPENCL] Eliminate usage of regex in opencl

### PR DESCRIPTION
**Pull request by omarahmed1111**
Usage of regex was downgrading the performance of opencl adapter strongly. Using [Intel PIN](https://www.intel.com/content/www/us/en/developer/articles/tool/pin-a-dynamic-binary-instrumentation-tool.html) tool, querying the total number of instructions using a custom pin tool, it reported that the affective number of instructions that was executed from inside `libpi_opencl.so` to run [simple-sycl-app](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#run-simple-dpc-application) was **269469** which is an order of magnitude higher than any other adapter. Analyzing this, most of these instructions was generated from the regex library. This was also definetly have a strong effect on the runtime performance of sycl-opencl applications.

The changes in this PR will decrease the effective number of instructions executed in opencl adapter to **~12000** which is a reasonable number comparing it to other adapters (all are in the range of **10000 - ~20000**).

### CODE REVIEW

**Comment by omarahmed1111** [comment]
Intel/llvm testing [here](https://github.com/intel/llvm/pull/11204)

File: source/adapters/opencl/kernel.cpp [hunk 1]
```diff
@@ -9,6 +9,8 @@
 //===----------------------------------------------------------------------===//
 #include "common.hpp"
 
+#include <memory>
```
Comment (aarongreig) [inline source/adapters/opencl/kernel.cpp:12]: what's this for? did including `<regex>` in common.hpp pull this in before it was removed?

**Review by aarongreig — APPROVED** [review]

File: source/adapters/opencl/kernel.cpp (hunk 1 above)
Comment (omarahmed1111) [reply source/adapters/opencl/kernel.cpp:12]: yes, the build was not seeing `shared_ptr` any more, I think that was because of regex was including this.

**Review by fabiomestre — APPROVED** [review]
LGTM

**Pull request merged**